Move few GivenWeWantToRequireWindowsForDesktopApps tests

wli3 opened the pull request:
Fix https://github.com/dotnet/installer/issues/8203

joeloff commented on test/Microsoft.DotNet.Tools.Tests.Utilities/PlatformSpecificFact.cs:
```diff
@@ -0,0 +1,44 @@
+        tvOS = 256,
+        Android = 512,
+        Browser = 1024,
+        AnyUnix = 2046
```
Should AnyUnix be 2048?

wli3 replied:
I just copied it from microsoft.extension.xunit. Since there is no point to take a dependency just for a enum. I will change this to 2048.

joeloff replied:
Yeah, thinking more about it, since this is defined as Flags, I'd expect AnyUnix to just be a bitmask of all the *nix platforms ORed, unless its intended to mean "any other unix not listed"

dsplaisted commented:
@wli3 Did we need to move this test, or could we have just added it to the [list of SDK tests that are run in the installer repo](https://github.com/dotnet/installer/blob/master/test/SdkTests/TestConfig.xml)?

wli3 commented:
@dsplaisted I moved the test. Since it will never pass in dotnet/sdk now. Since I added the change to override BundledNETCoreAppPackageVersion, and dotnet/sdk will need to download the latest targeting pack.
